Source/Scene/PointCloudEyeDomeLighting.js

```diff
@@ -0,0 +1,400 @@
+                'void main() \n' +
+                '{ \n' +
+                '    czm_point_cloud_post_process_main(); \n' +
+                '    v_positionECPS = (czm_inverseProjection * gl_Position).xyz; \n' +
+                '}');
+            fs.sources.unshift('#extension GL_EXT_draw_buffers : enable \n');
+            fs.sources.push(
+                'varying vec3 v_positionECPS; \n' +
+                'void main() \n' +
+                '{ \n' +
+                '    czm_point_cloud_post_process_main(); \n' +
+                // Write log base 2 depth to alpha for EDL
+                '    gl_FragData[1] = vec4(v_positionECPS, log2(-v_positionECPS.z)); \n' +
+                '}');
+
+            shader = context.shaderCache.createDerivedShaderProgram(shaderProgram, 'EC', {
+                vertexShaderSource : vs,
+                fragmentShaderSource : fs,
+                attributeLocations : attributeLocations
+            });
+        }
+
+        return shader;
+    }
+
+    PointCloudEyeDomeLighting.prototype.update = function(frameState, commandStart, tileset) {
+        var passes = frameState.passes;
+        var isPick = (passes.pick && !passes.render);
+        if (!processingSupported(frameState.context) || isPick) {
+            return;
+        }
+
+        this._strength = tileset.pointShading.eyeDomeLightingStrength;
+        this._radius = tileset.pointShading.eyeDomeLightingRadius;
+
+        var dirty = createResources(this, frameState.context, false);
+
+        // Hijack existing point commands to render into an offscreen FBO.
+        var i;
+        var commandList = frameState.commandList;
+        var commandEnd = commandList.length;
+
+        for (i = commandStart; i < commandEnd; ++i) {
+            var command = commandList[i];
+            if (command.primitiveType !== PrimitiveType.POINTS) {
+                continue;
+            }
+            var derivedCommand = command.derivedCommands.pointCloudProcessor;
+            if (!defined(derivedCommand) || command.dirty || dirty ||
+                (derivedCommand.framebuffer !== this._framebuffers.prior)) { // Prevent crash when tiles out-of-view come in-view during context size change
+                derivedCommand = DrawCommand.shallowClone(command);
+                command.derivedCommands.pointCloudProcessor = derivedCommand;
+
+                derivedCommand.framebuffer = this._framebuffers.prior;
+                derivedCommand.shaderProgram = getECShaderProgram(frameState.context, command.shaderProgram);
+                derivedCommand.castShadows = false;
+                derivedCommand.receiveShadows = false;
+
+                var derivedCommandRenderState = clone(derivedCommand.renderState);
+                derivedCommand.renderState = RenderState.fromCache(
+                    derivedCommandRenderState
+                );
+
+                derivedCommand.pass = Pass.CESIUM_3D_TILE; // Overrides translucent commands
```

Comment:
Out of curiosity, have you tried EDL with translucent points and seen if it works?

You can try `cesium\Specs\Data\Cesium3DTiles\PointCloud\PointCloudRGBA`

Reply:
Unfortunately this doesn't seem to work... also, for some reason `cesium\Specs\Data\Cesium3DTiles\PointCloud\PointCloudRGB` draws strangely with picking and EDL. I'll try to figure out what's wrong for `RGB` and see if we can detect transparency for cases like `RGBA`.

Reply:
Also, this is weird and I should figure it out:
![weird](https://user-images.githubusercontent.com/5395255/35535407-7056edc8-0511-11e8-8183-f253211bd6a6.JPG)